core/ir/aarch64/codec.c

```diff
@@ -5193,9 +5203,20 @@ encode_opnd_svemem_gpr_simm6_vl(uint enc, int opcode, byte *pc, opnd_t opnd,
     if (!opnd_is_base_disp(opnd) || opnd_get_index(opnd) != DR_REG_NULL ||
         opnd_get_size(opnd) != mem_transfer)
         return false;
+    if (!reg_is_gpr(opnd_get_base(opnd)))
+        return false;
+
+    /* As described in decode_opnd_svemem_gpr_simm6_vl(), disp is a multiple of
+     * vector length at the IR level, transformed to a vector index in the
+     * encoding.
+     */
+    int vl_bytes = dr_get_sve_vl() / 8;
+    ASSERT((opnd_get_disp(opnd) % vl_bytes) == 0);
```

Comment:
Best to just return false and fail the encoding like all the other un-encodable cases checked in this routine, rather than assert, if it's not an even multiple.

Reply:
Done.